js/lex.go

```diff
@@ -289,35 +288,35 @@ func (l *Lexer) consumeUnicodeEscape() bool {
 	return true
 }
 
-////////////////////////////////////////////////////////////////
-
-func (l *Lexer) consumeSingleLineToken(skip int) {
-	l.r.Move(skip)
+func (l *Lexer) consumeSingleLineComment() {
 	for {
 		c := l.r.Peek(0)
 		if c == '\r' || c == '\n' || c == 0 {
 			break
 		} else if c >= 0xC0 {
-			mark := l.r.Pos()
+			//mark := l.r.Pos()
 			if r, _ := l.r.PeekRune(0); r == '\u2028' || r == '\u2029' {
-				l.r.Rewind(mark)
+				//l.r.Rewind(mark)
 				break
 			}
 		}
 		l.r.Move(1)
```

Comment:
Unrelated to the changes, but I wonder how this will work for runes that are not end-of-line. Shouldn't we skip entire rune instead of a single byte in order not to appear in the middle of unicode character?